### DISCUSSION TITLE
Is TLS1.3 supported?

### DISCUSSION THREAD

**Discussion by simongurney1** [dropwizard/dropwizard#5272] [Q&A] [answered]
All code and documentation that I have looked through (such as https://www.dropwizard.io/en/latest/manual/configuration.html#tls) don't mention TLS1.3.

If no supportedProtocols key is provided, will dropwizard negotiate TLS1.3 if it is available?

**Comment by joschi** [comment, marked as answer]
Whether TLS 1.3 is being supported and used is a capability of your runtime platform, i. e. the version of the Java Runtime Environment you are using.

If your JRE supports TLS 1.3 and `tls.protocol` has been set to `TLSv1.3`, then Dropwizard (or rather the HTTP client) will use it.

**Reply by simongurney1** [reply]
Thanks.  So if tls.protocol is set to TLS1.2, but we do not provide a supportedProtocols key, would TLS1.3 be used if the other end mandated it or is tls.protocol the maximum if its set?

**Reply by joschi** [reply]
It will use TLSv1.2 as default and switch to any of the supported protocols if the server only supports these.

- https://github.com/dropwizard/dropwizard/blob/v2.1.0/dropwizard-client/src/main/java/io/dropwizard/client/DropwizardSSLConnectionSocketFactory.java#L77-L78
- https://github.com/dropwizard/dropwizard/blob/v2.1.0/dropwizard-client/src/main/java/io/dropwizard/client/DropwizardSSLConnectionSocketFactory.java#L44-L45
- https://www.javadoc.io/doc/org.apache.httpcomponents/httpclient/4.5.13/org/apache/http/conn/ssl/SSLConnectionSocketFactory.html#SSLConnectionSocketFactory(javax.net.ssl.SSLContext,%20java.lang.String[],%20java.lang.String[],%20javax.net.ssl.HostnameVerifier)
- https://javadoc.io/doc/org.apache.httpcomponents/httpcore/4.4.15/org/apache/http/ssl/SSLContextBuilder.html#setProtocol(java.lang.String)
